p2p/host/resource-manager/conn_limiter.go

```diff
@@ -90,18 +95,43 @@ func WithNetworkPrefixLimit(ipv4 []NetworkPrefixLimit, ipv6 []NetworkPrefixLimit
 // limit for any given subnet.
 func WithLimitPerSubnet(ipv4 []ConnLimitPerSubnet, ipv6 []ConnLimitPerSubnet) Option {
 	return func(rm *resourceManager) error {
+		connLimiterInstance, ok := rm.connLimiter.(*connLimiter)
+		if !ok {
+			return nil // early exit, it might be a custom implementation
+		}
+
 		if ipv4 != nil {
-			rm.connLimiter.connLimitPerSubnetV4 = ipv4
+			connLimiterInstance.connLimitPerSubnetV4 = ipv4
```

Comment:
we need to protect this set method with the same connLimiter.mu mutex. Maybe extract these set methods in the implementation + interface and get rid of the casts?